### PR TITLE
Let evaluator add name of builtin throwing exception

### LINKED ISSUE #108: Let evaluator add name of builtin throwing builtin-ex
To mimic the error messages from OPA, we prepend any built-in exception with the name of the builtin throwing the exception, like:

```clojure
(throw (builtin-ex "graph.reachable: some error message"))
```
Since some exception conditions are shared between more than one builtin, this leads to repetition, and feels like a hard coupling we'd be better off without. Since the evaluator knows which builtin is throwing an exception, we could just have that add the name of the builtin when caught and rethrown.

### PR DESCRIPTION
**Pull request by anderseknert**
Fixes #108

Signed-off-by: Anders Eknert <anders@eknert.com>

### CODE REVIEW

File: src/main/cljc/jarl/eval.cljc [hunk 1]
```diff
@@ -488,7 +488,10 @@
             (do
               (log/tracef "function <%s> threw error: %s" name (ex-message e))
               (if (true? (get state :strict-builtin-errors))
-                (throw e)
+                (let [[rego-type & m] (str/split (ex-message e) #":")
+                      with-name (str/join ": " (apply conj [rego-type name] (mapv str/trim m)))
```
Comment (johanfylling) [inline src/main/cljc/jarl/eval.cljc:492]: I might be reading this wrong. But if an error contains the string `eval_builtin_error: invalid string: "foo : bar:baz"`, won't that get transformed to `eval_builtin_error: <some-builtin>: invalid string: "foo: bar: baz"`, changing parts of the message that should remain intact?

Perhaps we could just strip out the type prefix from the message (it's always included as `::type` on the `ex-info`, right?) and just `(str/join type-from-ex-info ": " name ": " message-without-type-prefix)`?

File: src/main/cljc/jarl/eval.cljc [hunk 2]
```diff
@@ -488,7 +488,10 @@
             (do
               (log/tracef "function <%s> threw error: %s" name (ex-message e))
               (if (true? (get state :strict-builtin-errors))
-                (throw e)
+                (let [[rego-type & m] (str/split (ex-message e) #":")
```
Comment (johanfylling) [inline src/main/cljc/jarl/eval.cljc:491]: Could be moved to a helper function in `jarl.exceptions`, I think. Something like `jarl.exceptions/add-builtin-name`, that takes the exception and builtin name as args?

File: src/main/cljc/jarl/eval.cljc (hunk 1 above)
Comment (anderseknert) [reply src/main/cljc/jarl/eval.cljc:492]: Sounds good 👍  I'll take a look.

Comment (anderseknert) [reply src/main/cljc/jarl/eval.cljc]: @johanfylling just pushed - PTAL :)

File: src/main/cljc/jarl/eval.cljc (hunk 2 above)
Comment (anderseknert) [reply src/main/cljc/jarl/eval.cljc]: With the above change, this is only one line of code anyway, so might just keep it like it is unless we need to do this in more places later.

**Review by johanfylling — APPROVED** [review]
👍

**Pull request merged**